quarkchain/evm/messages.py

```diff
@@ -351,7 +361,31 @@ def apply_transaction(state, tx: transactions.Transaction, tx_wrapper_hash):
         state.get_balance(tx.sender, token_id=tx.gas_token_id)
         >= tx.startgas * tx.gasprice
     )
-    state.delta_token_balance(tx.sender, tx.gas_token_id, -tx.startgas * tx.gasprice)
+    original_gas_token_id = tx.gas_token_id
+    refund_rate = 100
+    if tx.gas_token_id != state.genesis_token_id:
```

Comment:
I don't think the code will work here, because only `ShardState` has `genesis_token_id`. can you add some unit tests?